miasm/expression/simplifications_common.py

```diff
@@ -1554,3 +1553,25 @@ def simp_add_multiple(_, expr):
     if len(out) == 1:
         return out[0]
     return ExprOp('+', *out)
+
+def simp_compose_and_mask(_, expr):
+    """
+    {X 0 8, Y 8 16} & 0xFF => X
+    {X 0 32} & 0xFFFF => X[0:16]
+    {X 0 8, Y 8 24, Z 24 32} & 0xFFFFFF => X|Y
+    """
+    if not expr.is_op('&'):
+        return expr
+    # handle the case where arg2 = arg1.mask
+    if len(expr.args) != 2:
+        return expr
+    arg1, arg2 = expr.args
+    if not arg1.is_compose():
+        return expr
+    if not arg2.is_int():
+        return expr
+    int2 = arg2.arg.arg
+    if (int2 + 1) & int2 != 0:
```

Comment:
Take care here:
- arg2 is an Expr
- arg2.arg is a moduint
- arg2.arg.arg is a python int

So in python int, 0xFFFFFFFF + 1 = 0x100000000
```python
>>> cst = ExprInt(0xFFFFFFFF, 32)
>>> print cst
0xFFFFFFFF
>>> print cst.arg
uint32(0xffffffff)
>>> print cst.arg.arg
4294967295
>>> print cst.arg.arg + 1
4294967296
>>>  
```

*but*
```python
>>> print cst.arg + 1
uint32(0x0)
>>> 
```
Maybe you want the second case here

Reply:
Hum. Ok no. it works with and without the modulus computation.
So maybe you can just have:
```pyhton
int2 = int(arg2)
```
instead of:
```python
int2 = arg2.arg.arg
```

Reply:
Didn't see your latest comment earlier so I pushed a new commit replacing the python int by the moduint, do you want me to make the change with `int(arg2)`?

Reply:
Yes please: and sorry for the bad comment :disappointed:

Reply:
No problem :)